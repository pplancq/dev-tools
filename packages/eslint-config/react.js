module.exports = {
  extends: ['./rules/react.js', './rules/prettier.js'].map(require.resolve),
};
