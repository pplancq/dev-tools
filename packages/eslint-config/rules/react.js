module.exports = {
  extends: [
    'airbnb',
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    './base.js',
    './typescript.js',
  ],
  rules: {
    // eslint-plugin-import https://github.com/import-js/eslint-plugin-import
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
      },
    ],

    // eslint-plugin-react https://github.com/jsx-eslint/eslint-plugin-react
    'react/no-multi-comp': [
      'error',
      {
        ignoreStateless: false,
      },
    ],
    'react/jsx-props-no-spreading': 'off',
    'react/no-unescaped-entities': 'off',
    'react/jsx-filename-extension': [
      'error',
      {
        extensions: ['.jsx'],
      },
    ],
    'react/function-component-definition': [
      'error',
      {
        namedComponents: ['function-declaration', 'arrow-function', 'function-expression'],
        unnamedComponents: ['arrow-function', 'function-expression'],
      },
    ],

    // eslint-plugin-react-hooks https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hook
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    react: {
      version: 'detect',
    },
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx'],
      },
    },
  },
  overrides: [
    {
      files: ['**/*.ts?(x)'],
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
      },
      rules: {
        // eslint-plugin-import https://github.com/import-js/eslint-plugin-import
        'import/extensions': [
          'error',
          'ignorePackages',
          {
            js: 'never',
            jsx: 'never',
            ts: 'never',
            tsx: 'never',
          },
        ],

        // eslint-plugin-react https://github.com/jsx-eslint/eslint-plugin-react
        'react/require-default-props': 'off',
        'react/jsx-filename-extension': [
          'error',
          {
            extensions: ['.jsx', '.tsx'],
          },
        ],
      },
    },
    {
      extends: ['plugin:jest-dom/recommended'],
      files: ['*.{test,spec,steps}.{js,jsx,ts,tsx}'],
      plugins: ['testing-library'],
      rules: {
        // eslint-plugin-testing-library https://github.com/testing-library/eslint-plugin-testing-library
        'testing-library/await-async-queries': 'error',
        'testing-library/await-async-utils': 'error',
        'testing-library/no-await-sync-queries': 'error',
        'testing-library/no-container': 'error',
        'testing-library/no-debugging-utils': 'error',
        'testing-library/no-dom-import': ['error', 'react'],
        'testing-library/no-node-access': 'error',
        'testing-library/no-promise-in-fire-event': 'error',
        'testing-library/no-render-in-lifecycle': 'error',
        'testing-library/no-unnecessary-act': 'error',
        'testing-library/no-wait-for-multiple-assertions': 'error',
        'testing-library/no-wait-for-side-effects': 'error',
        'testing-library/no-wait-for-snapshot': 'error',
        'testing-library/prefer-find-by': 'error',
        'testing-library/prefer-presence-queries': 'error',
        'testing-library/prefer-query-by-disappearance': 'error',
        'testing-library/prefer-screen-queries': 'error',
        'testing-library/render-result-naming-convention': 'error',
      },
    },
  ],
};
