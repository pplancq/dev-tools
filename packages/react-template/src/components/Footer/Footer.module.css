.container {
  width: auto;
  padding: 16px;
}
