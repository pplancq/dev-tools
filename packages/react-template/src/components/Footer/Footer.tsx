import logo from '@Front/assets/images/logo.png';
import { Logo } from '@Front/ui/atoms/Logo';
import { Link } from 'react-router';

import classes from './Footer.module.css';

export const Footer = () => {
  return (
    <div className={classes.container}>
      <hr />
      <Link to="/">
        <Logo size="50px" src={logo} alt="logo" />
      </Link>
    </div>
  );
};
