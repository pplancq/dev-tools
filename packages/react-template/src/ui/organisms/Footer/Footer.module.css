footer {
  width: 100%;
  flex-grow: 0;
  flex-shrink: 0;
  letter-spacing: 4px;
  text-align: center;
  text-transform: uppercase;
  color: white;
  background-color: #00008f;
}
