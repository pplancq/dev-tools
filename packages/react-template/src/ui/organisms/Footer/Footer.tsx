import { ReactNode } from 'react';
import classes from './Footer.module.css';

type FooterProps = {
  footerContent?: ReactNode;
};

export const Footer = ({ footerContent }: FooterProps) => {
  return <footer className={classes.footer}>{footerContent}</footer>;
};
