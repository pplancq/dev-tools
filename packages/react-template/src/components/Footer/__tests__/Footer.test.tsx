import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router';
import { Footer } from '../Footer';

describe('Footer Component', () => {
  it('should render the Footer component correctly', () => {
    const { container } = render(
      <Router>
        <Footer />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });
});
