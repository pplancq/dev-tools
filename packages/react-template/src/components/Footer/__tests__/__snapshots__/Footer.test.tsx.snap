// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > should render the Footer component correctly 1`] = `
<div>
  <div
    class="_container_acfd2c"
  >
    <hr />
    <a
      data-discover="true"
      href="/"
    >
      <img
        alt="logo"
        class="_logoStyle_9e86a7"
        src="/src/assets/images/logo.png"
        style="width: 50px;"
      />
    </a>
  </div>
</div>
`;
